src/app/(sidebar)/transaction/sign/page.tsx

```diff
@@ -1,5 +1,198 @@
+    undefined,
+  );
+  const [isTxImported, setIsTxImported] = useState<boolean>(false);
+
+  const onChange = (value: string) => {
+    setTxErrMsg("");
+    setTxSuccessMsg("");
+    setTxEnv(value);
+
+    if (value.length > 0) {
+      const validatedXDR = validate.xdr(value);
+
+      if (validatedXDR.result === "success") {
+        setIsTxValid(true);
+        setTxSuccessMsg(validatedXDR.message);
+      } else {
+        setIsTxValid(false);
+        setTxErrMsg(validatedXDR.message);
+      }
+    }
+  };
+
+  const onImport = () => {
+    try {
+      const transaction = TransactionBuilder.fromXDR(txEnv, network.passphrase);
+      setIsTxImported(true);
+      setTx(transaction);
+    } catch (e) {
+      setIsTxImported(false);
+      setTxErrMsg("Unable to import a transaction envelope");
+    }
+  };
+
+  const rendeImportView = () => {
+    return (
+      <>
+        <Card>
+          <div className="SignTx__xdr">
+            <XdrPicker
+              id="sign-transaction-xdr"
+              label={
+                <Text size="xs" as="span" weight="medium">
+                  Import a transaction envelope in XDR format
+                  <span className="SignTx__icon">
+                    <Icon.AlertCircle />
+                  </span>
+                </Text>
+              }
+              value={txEnv || ""}
+              error={txErrMsg}
+              note={
+                <Text size="xs" as="span" addlClassName="success-message">
+                  {txSuccessMsg}
+                </Text>
+              }
+              onChange={(e) => onChange(e.target.value)}
+            />
+
+            <div className="SignTx__CTA">
+              <Button
+                disabled={!txEnv || !isTxValid}
+                size="md"
+                variant={"secondary"}
+                onClick={onImport}
+              >
+                Import transaction
+              </Button>
+            </div>
+          </div>
+        </Card>
+
+        <Alert
+          placement="inline"
+          variant="primary"
+          actionLink="https://developers.stellar.org/network/horizon/resources"
+          actionLabel="Read more about signatures on the developer's site"
+        >
+          <Text size="sm" as="p">
+            The transaction signer lets you add signatures to a Stellar
+            transaction. Signatures are used in the network to prove that the
+            account is authorized to perform the operations in the transaction.
+          </Text>
+          <Text size="sm" as="p">
+            For simple transactions, you only need one signature from the
+            correct account. Some advanced transactions may require more than
+            one signature if there are multiple source accounts or signing keys.
+          </Text>
+        </Alert>
+      </>
+    );
+  };
+
+  const renderOverviewView = () => {
+    if (!tx) {
+      return;
```

Comment:
React might not be happy if we return `undefined`

```suggestion
      return null;
```